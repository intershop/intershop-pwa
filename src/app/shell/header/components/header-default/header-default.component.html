<ng-template #searchBoxTemplate let-icon="icon">
  <ish-search-box-container
    data-testing-id="search-box-desktop"
    [configuration]="{
      id: 'header',
      placeholder: 'search.searchbox.instructional_text' | translate,
      autoSuggest: true,
      maxAutoSuggests: 10,
      icon: icon,
      showLastSearchTerm: true
    }"
  ></ish-search-box-container>
</ng-template>

<div class="header container">
  <ul *ngIf="!isSticky" class="user-links">
    <li *ngIf="!isSticky && deviceType !== 'mobile'">
      <ish-login-status-container data-testing-id="user-status-desktop"></ish-login-status-container>
    </li>
    <li *ishFeature="'compare'" class="d-none d-md-block">
      <ish-product-compare-status-container data-testing-id="compare-status-desktop">
      </ish-product-compare-status-container>
    </li>
  </ul>

  <div class="header-utility">
    <a class="search-toggler" [ngClass]="{ 'active-search': showSearch }">
      <fa-icon *ngIf="isSticky" class="header-icon" [icon]="['fas', 'search']" (click)="toggle('search')"></fa-icon>
      <div
        *ngIf="isSticky || showSearch"
        class="search-container header-search-container"
        [ngbCollapse]="!showSearch"
        [ngClass]="{ show: !isSticky }"
      >
        <ng-container *ngTemplateOutlet="searchBoxTemplate; context: { icon: 'angle-right' }"></ng-container>
      </div>
    </a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container *ngIf="!isSticky" data-testing-id="language-switch-desktop">
      </ish-language-switch-container>
    </div>
    <ish-login-status-container *ngIf="isSticky || deviceType === 'mobile'" view="small"></ish-login-status-container>

    <ish-mini-basket-container
      data-testing-id="mini-basket-desktop"
      view="full"
      class="d-none"
      [ngClass]="isSticky ? 'd-lg-block' : 'd-md-block'"
    ></ish-mini-basket-container>
    <ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      [ngClass]="isSticky ? 'd-lg-none' : 'd-md-none'"
    ></ish-mini-basket-container>
  </div>

  <div *ngIf="!isSticky && deviceType !== 'mobile'" class="search-container header-search-container">
    <ng-container *ngTemplateOutlet="searchBoxTemplate"></ng-container>
  </div>

  <div class="logo-wrapper">
    <a
      *ngIf="showDesktopLogoLink"
      rel="home"
      routerLink="/home"
      class="logo"
      data-testing-id="header-home-link-desktop"
    >
      <img src="assets/img/logo.png" alt="Logo" />
    </a>
    <a
      *ngIf="showMobileLogoLink"
      rel="home"
      routerLink="/home"
      class="mobile-logo"
      data-testing-id="header-home-link-mobile"
      ><img src="assets/img/logo_mobile.png" alt="Logo" width="41" height="41"
    /></a>
  </div>

  <button class="navbar-toggler" type="button" (click)="toggle('navbar')">
    <span class="sr-only">{{ 'common.button.navbarCollapsed.text' | translate }}</span>
    <div class="hamburger-menu" [ngClass]="{ open: showNavBar }">
      <!--- strokes of the hamburger menu -->
      <span></span> <span></span> <span></span>
    </div>
  </button>

  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" [ngbCollapse]="!showNavBar">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container>
        <ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
