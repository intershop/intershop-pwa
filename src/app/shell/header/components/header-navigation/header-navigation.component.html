<ul class="navbar-nav main-navigation-list d-md-none" [ngClass]="{ accordion: view !== 'full' }">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<ul class="navbar-nav main-navigation-list d-none d-md-block" [ngClass]="{ accordion: view === 'small' }">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<!--- Category Template -->
<ng-template #categoryTemplate>
  <li
    *ngFor="let category of categories"
    #submenu
    class="dropdown"
    [ngClass]="{ open: isOpened(category) }"
    (mouseover)="subMenuShow(submenu)"
    (mouseout)="subMenuHide(submenu)"
    (click)="subMenuHide(submenu)"
  >
    <a
      [routerLink]="['/category', category.uniqueId]"
      [attr.data-testing-id]="category.uniqueId + '-link'"
      [ngStyle]="{ width: !category.hasChildren() ? '100%' : '' }"
    >
      {{ category.name }}
    </a>
    <a
      *ngIf="category.hasChildren()"
      class="dropdown-toggle"
      [ngClass]="{ 'd-none': view === 'full', 'd-md-none': view === 'auto' }"
      (click)="toggleOpen(category)"
    >
      <fa-icon *ngIf="isOpened(category); else closed" [icon]="['fas', 'minus']"></fa-icon>
      <ng-template #closed> <fa-icon [icon]="['fas', 'plus']"></fa-icon> </ng-template>
    </a>
    <ish-sub-category-navigation
      [category]="category"
      [subCategoriesDepth]="1"
      [view]="view"
    ></ish-sub-category-navigation>
  </li>
</ng-template>
