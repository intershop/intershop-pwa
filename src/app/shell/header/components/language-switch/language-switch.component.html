<div *ngIf="locale" class="language-switch" [ngClass]="view" ngbDropdown placement="bottom-right">
  <a ngbDropdownToggle class="language-switch-link">
    <fa-icon [icon]="['fas', 'globe-americas']" class="header-icon"></fa-icon>
    <span class="language-switch-current-selection d-inline">{{ view==="accordion"?locale.displayName: locale.value }}<span class="switch_arrow"></span></span>
  </a>
  <div ngbDropdownMenu class="language-switch-container dropdown-menu" style="left:0px!important">
    <div class="language-switch-menu-container">
      <ul *ngIf="availableLocales && availableLocales.length">
        <ng-container *ngFor="let l of availableLocales">
          <li *ngIf="l.lang !== locale.lang" >
          <a (click)="switch(l)">
            {{ l.displayName }}
          </a>
        </li>
        </ng-container>
      </ul>
    </div>
  </div>
</div>
