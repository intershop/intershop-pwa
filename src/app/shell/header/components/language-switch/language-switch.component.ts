import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';

import { Locale } from 'ish-core/models/locale/locale.model';

@Component({
  selector: 'ish-language-switch',
  templateUrl: './language-switch.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class LanguageSwitchComponent {
  @Input() locale: Locale;
  @Input() availableLocales: Locale[];
  @Input() view: '' | 'accordion' = '';
  @Output() localeChange = new EventEmitter<Locale>();

  switch(locale: Locale) {
    this.localeChange.emit(locale);
    return false; // prevent actual navigation, only change localized values on the current page and set language state
  }
}
