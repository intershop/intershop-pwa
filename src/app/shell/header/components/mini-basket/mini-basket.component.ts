import { Location } from '@angular/common';
import { ChangeDetectionStrategy, Component, Input, OnChanges, SimpleChanges } from '@angular/core';

import { BasketView } from 'ish-core/models/basket/basket.model';
import { HttpError } from 'ish-core/models/http-error/http-error.model';

/**
 * The Mini Basket Component displays a quick overview over the users basket items.
 * It uses the {@link ProductImageComponent} for the rendering of product images.
 *
 * @example
 * <ish-mini-basket
      [basket]="basket$ | async"
      [view]="view"
      [basketAnimation]="basketAnimation$ | async"
      [error]="basketError$ | async"
    ></ish-mini-basket>
 */
@Component({
  selector: 'ish-mini-basket',
  templateUrl: './mini-basket.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class MiniBasketComponent implements OnChanges {
  /**
   * The basket that should be displayed.
   */
  @Input() basket: BasketView;
  @Input() view: 'auto' | 'small' | 'full' = 'auto';
  @Input() basketAnimation = '';
  @Input() error: HttpError;

  isCollapsed = true;
  itemCount = 0;

  constructor(private location: Location) {}

  ngOnChanges(c: SimpleChanges) {
    this.animateBasket(c);
    if (this.basket) {
      this.itemCount = this.basket.totalProductQuantity;
    } else {
      this.itemCount = 0;
    }
  }

  /**
   * Animate basket after the product line item count has changed
   * Open basket if an error occured
   */
  animateBasket(c: SimpleChanges) {
    if (c && c.basketAnimation && this.location.path() !== '/basket') {
      if (this.basketAnimation && c.basketAnimation.currentValue === 'tada') {
        this.open();
      } else {
        this.collapse();
      }
    }
    if (c && !!c.error && this.error) {
      this.open();
    }
  }

  /**
   * Toggle the collapse state of the mini basket programmatically.
   */
  toggleCollapse() {
    this.isCollapsed = !this.isCollapsed;
  }

  /**
   * Collapse the mini basket programmatically.
   */
  collapse() {
    this.isCollapsed = true;
  }

  /**
   * Open the mini basket programmatically.
   */
  open() {
    this.isCollapsed = false;
  }
}
