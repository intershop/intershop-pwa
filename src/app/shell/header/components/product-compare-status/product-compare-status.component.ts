import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'ish-product-compare-status',
  templateUrl: './product-compare-status.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductCompareStatusComponent {
  @Input() productCompareCount: number;

  @Input() view: 'auto' | 'small' | 'full' = 'auto';
}
