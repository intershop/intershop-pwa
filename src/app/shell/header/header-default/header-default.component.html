<ng-template #searchBoxTemplate let-icon="icon">
  <ish-lazy-search-box
    data-testing-id="search-box-desktop"
    [configuration]="{
      placeholder: 'search.searchbox.instructional_text' | translate,
      autoSuggest: true,
      maxAutoSuggests: 10,
      icon: icon,
      showLastSearchTerm: true
    }"
  ></ish-lazy-search-box>
</ng-template>

<div class="header container" [ngClass]="deviceType">
  <ul *ngIf="!isSticky" class="user-links">
    <li *ngIf="!isSticky && deviceType !== 'mobile'">
      <ish-login-status data-testing-id="user-status-desktop"></ish-login-status>
    </li>
    <li *ishFeature="'compare'" class="d-none d-md-block">
      <ish-product-compare-status data-testing-id="compare-status-desktop"> </ish-product-compare-status>
    </li>
    <li *ishFeature="'wishlists'" class="d-none d-md-block">
      <ish-lazy-wishlists-link></ish-lazy-wishlists-link>
    </li>
  </ul>

  <div class="header-utility">
    <a class="search-toggler" [ngClass]="{ 'active-search': showSearch }">
      <fa-icon *ngIf="isSticky" class="header-icon" [icon]="['fas', 'search']" (click)="toggle('search')"></fa-icon>
      <div
        *ngIf="isSticky || showSearch"
        class="search-container header-search-container"
        [ngbCollapse]="!showSearch"
        [ngClass]="{ show: !isSticky }"
      >
        <ng-container *ngTemplateOutlet="searchBoxTemplate; context: { icon: 'angle-right' }"></ng-container>
      </div>
    </a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch *ngIf="!isSticky" data-testing-id="language-switch-desktop"> </ish-language-switch>
    </div>
    <ish-login-status *ngIf="isSticky || deviceType === 'mobile'" view="small"></ish-login-status>

    <ish-mini-basket
      data-testing-id="mini-basket-desktop"
      view="full"
      class="d-none"
      [ngClass]="isSticky ? 'd-lg-block' : 'd-md-block'"
    ></ish-mini-basket>
    <ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      [ngClass]="isSticky ? 'd-lg-none' : 'd-md-none'"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          *ngIf="showDesktopLogoLink"
          rel="home"
          routerLink="/home"
          class="logo"
          data-testing-id="header-home-link-desktop"
        ></a>
        <a
          *ngIf="showMobileLogoLink"
          rel="home"
          routerLink="/home"
          class="mobile-logo"
          data-testing-id="header-home-link-mobile"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right">
      <ish-lazy-header-quickorder></ish-lazy-header-quickorder>
    </div>
    <div class="col-md-4">
      <div *ngIf="!isSticky && deviceType !== 'mobile'" class="search-container header-search-container">
        <ng-container *ngTemplateOutlet="searchBoxTemplate"></ng-container>
      </div>
    </div>
  </div>

  <button class="navbar-toggler" type="button" (click)="toggle('navbar')">
    <span class="sr-only">{{ 'common.button.navbarCollapsed.text' | translate }}</span>
    <div class="hamburger-menu" [ngClass]="{ open: showNavBar }">
      <!--- strokes of the hamburger menu -->
      <span></span> <span></span> <span></span>
    </div>
  </button>

  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" [ngbCollapse]="!showNavBar">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation>
        <ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
