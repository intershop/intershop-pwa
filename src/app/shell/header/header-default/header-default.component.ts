import { ChangeDetectionStrategy, Component, Input, OnChanges, SimpleChanges } from '@angular/core';

import { DeviceType } from 'ish-core/models/viewtype/viewtype.types';

type CollapsibleComponent = 'search' | 'navbar' | 'minibasket';

/**
 * The Header Component displays the page header.
 *
 * It uses the {@link LoginStatusContainerComponent} for rendering the users login status.
 * It uses the {@link ProductCompareStatusContainerComponent} for rendering the product compare button and count.
 * It uses the {@link MobileBasketContainerComponent} for rendering the mobile basket button and basket item count.
 * It uses the {@link LanguageSwitchContainerComponent} for rendering the language selection dropdown.
 * It uses the {@link SearchBoxContainerComponent} for rendering the search box.
 * It uses the {@link HeaderNavigationContainerComponent} for rendering the pages main navigation.
 * It uses the {@link MiniBasketContainerComponent} for rendering mini basket on desktop sized viewports.
 *
 * @example
 * <ish-header></ish-header>
 */
@Component({
  selector: 'ish-header-default',
  templateUrl: './header-default.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HeaderDefaultComponent implements OnChanges {
  @Input() isSticky = false;
  @Input() deviceType: DeviceType;
  @Input() reset: unknown;

  activeComponent: CollapsibleComponent = 'search';

  ngOnChanges(changes: SimpleChanges) {
    if (changes.reset) {
      this.activeComponent = 'search';
    }
    this.toggleSpecialStatusOfSearch();
  }

  get showSearch() {
    return (
      this.activeComponent === 'search' &&
      // always show for sticky header
      (this.deviceType === 'mobile' || this.isSticky)
    );
  }

  get showNavBar() {
    return (
      this.activeComponent === 'navbar' ||
      // always show for desktop
      this.deviceType === 'desktop' ||
      // always show for tablet on top
      (this.deviceType === 'tablet' && !this.isSticky)
    );
  }

  get showDesktopLogoLink() {
    return (!this.isSticky && this.deviceType === 'tablet') || this.deviceType === 'desktop';
  }

  get showMobileLogoLink() {
    return (this.isSticky && this.deviceType !== 'desktop') || this.deviceType === 'mobile';
  }

  private toggleSpecialStatusOfSearch() {
    // deactivate search when switching to sticky header
    if (this.isSticky && this.activeComponent === 'search') {
      this.activeComponent = undefined;
    }
    // activate search when scrolling to top and no other is active
    if (!this.isSticky && !this.activeComponent) {
      this.activeComponent = 'search';
    }
  }

  toggle(component: CollapsibleComponent) {
    if (this.activeComponent === component) {
      // activate search bar when on top and no other is active
      this.activeComponent = !this.isSticky ? 'search' : undefined;
    } else {
      this.activeComponent = component;
    }
  }
}
