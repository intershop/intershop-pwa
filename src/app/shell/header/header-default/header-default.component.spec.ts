import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FaIconComponent } from '@fortawesome/angular-fontawesome';
import { NgbCollapse } from '@ng-bootstrap/ng-bootstrap';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';

import { FeatureToggleModule } from 'ish-core/feature-toggle.module';
import { findAllCustomElements } from 'ish-core/utils/dev/html-query-utils';
import { HeaderNavigationComponent } from 'ish-shell/header/header-navigation/header-navigation.component';
import { LanguageSwitchComponent } from 'ish-shell/header/language-switch/language-switch.component';
import { LoginStatusComponent } from 'ish-shell/header/login-status/login-status.component';
import { MiniBasketComponent } from 'ish-shell/header/mini-basket/mini-basket.component';
import { ProductCompareStatusComponent } from 'ish-shell/header/product-compare-status/product-compare-status.component';
import { UserInformationMobileComponent } from 'ish-shell/header/user-information-mobile/user-information-mobile.component';
import { LazySearchBoxComponent } from 'ish-shell/shared/lazy-search-box/lazy-search-box.component';

import { LazyHeaderQuickorderComponent } from '../../../extensions/quickorder/exports/lazy-header-quickorder/lazy-header-quickorder.component';
import { LazyWishlistsLinkComponent } from '../../../extensions/wishlists/exports/lazy-wishlists-link/lazy-wishlists-link.component';

import { HeaderDefaultComponent } from './header-default.component';

describe('Header Default Component', () => {
  let fixture: ComponentFixture<HeaderDefaultComponent>;
  let element: HTMLElement;
  let component: HeaderDefaultComponent;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [FeatureToggleModule.forTesting('compare'), TranslateModule.forRoot()],
      declarations: [
        HeaderDefaultComponent,
        MockComponent(FaIconComponent),
        MockComponent(HeaderNavigationComponent),
        MockComponent(LanguageSwitchComponent),
        MockComponent(LazyHeaderQuickorderComponent),
        MockComponent(LazySearchBoxComponent),
        MockComponent(LazyWishlistsLinkComponent),
        MockComponent(LoginStatusComponent),
        MockComponent(MiniBasketComponent),
        MockComponent(NgbCollapse),
        MockComponent(ProductCompareStatusComponent),
        MockComponent(UserInformationMobileComponent),
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderDefaultComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
  });

  it('should render User Links on template', () => {
    fixture.detectChanges();
    expect(findAllCustomElements(element)).toIncludeAllMembers(['ish-login-status', 'ish-product-compare-status']);
  });
  it('should render Language Switch on template', () => {
    fixture.detectChanges();
    expect(findAllCustomElements(element)).toContain('ish-language-switch');
  });

  it('should render Search Box on template', () => {
    fixture.detectChanges();
    expect(findAllCustomElements(element)).toContain('ish-lazy-search-box');
  });

  it('should render Header Navigation on template', () => {
    fixture.detectChanges();
    expect(findAllCustomElements(element)).toContain('ish-header-navigation');
  });

  it('should render normal header adequately for mobile devices', () => {
    component.deviceType = 'mobile';
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });

  it('should render sticky header adequately for mobile devices', () => {
    component.deviceType = 'mobile';
    component.isSticky = true;
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });

  it('should render normal header adequately for tablet devices', () => {
    component.deviceType = 'tablet';
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });

  it('should render sticky header adequately for tablet devices', () => {
    component.deviceType = 'tablet';
    component.isSticky = true;
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });

  it('should render normal header adequately for desktop', () => {
    component.deviceType = 'desktop';
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });

  it('should render sticky header adequately for desktop', () => {
    component.deviceType = 'desktop';
    component.isSticky = true;
    fixture.detectChanges();

    expect(element).toMatchSnapshot();
  });
});
