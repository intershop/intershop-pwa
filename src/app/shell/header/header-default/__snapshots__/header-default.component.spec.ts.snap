// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Default Component should render normal header adequately for desktop 1`] = `
<div class="header container desktop" ng-reflect-ng-class="desktop">
  <ul class="user-links">
    <li><ish-login-status data-testing-id="user-status-desktop"></ish-login-status></li>
    <li class="d-none d-md-block">
      <ish-product-compare-status
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch data-testing-id="language-switch-desktop"></ish-language-switch>
    </div>
    <ish-mini-basket
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="logo"
          data-testing-id="header-home-link-desktop"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4">
      <div class="search-container header-search-container">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box>
      </div>
    </div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu open"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="false">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Default Component should render normal header adequately for mobile devices 1`] = `
<div class="header container mobile" ng-reflect-ng-class="mobile">
  <ul class="user-links">
    <li class="d-none d-md-block">
      <ish-product-compare-status
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler active-search"
      ><div class="search-container header-search-container show" ng-reflect-collapsed="false">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box></div
    ></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch data-testing-id="language-switch-desktop"></ish-language-switch>
    </div>
    <ish-login-status view="small" ng-reflect-view="small"></ish-login-status
    ><ish-mini-basket
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="mobile-logo"
          data-testing-id="header-home-link-mobile"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4"></div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Default Component should render normal header adequately for tablet devices 1`] = `
<div class="header container tablet" ng-reflect-ng-class="tablet">
  <ul class="user-links">
    <li><ish-login-status data-testing-id="user-status-desktop"></ish-login-status></li>
    <li class="d-none d-md-block">
      <ish-product-compare-status
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch data-testing-id="language-switch-desktop"></ish-language-switch>
    </div>
    <ish-mini-basket
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="logo"
          data-testing-id="header-home-link-desktop"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4">
      <div class="search-container header-search-container">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box>
      </div>
    </div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu open"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="false">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Default Component should render sticky header adequately for desktop 1`] = `
<div class="header container desktop" ng-reflect-ng-class="desktop">
  <div class="header-utility">
    <a class="search-toggler active-search"
      ><fa-icon class="header-icon" ng-reflect-icon="fas,search"></fa-icon>
      <div class="search-container header-search-container" ng-reflect-collapsed="false">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box></div
    ></a>
    <div class="language-switch-container d-none d-md-block"></div>
    <ish-login-status view="small" ng-reflect-view="small"></ish-login-status
    ><ish-mini-basket
      class="d-none d-lg-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-lg-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-lg-none"
      ng-reflect-ng-class="d-lg-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="logo"
          data-testing-id="header-home-link-desktop"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4"></div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu open"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="false">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Default Component should render sticky header adequately for mobile devices 1`] = `
<div class="header container mobile" ng-reflect-ng-class="mobile">
  <div class="header-utility">
    <a class="search-toggler active-search"
      ><fa-icon class="header-icon" ng-reflect-icon="fas,search"></fa-icon>
      <div class="search-container header-search-container" ng-reflect-collapsed="false">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box></div
    ></a>
    <div class="language-switch-container d-none d-md-block"></div>
    <ish-login-status view="small" ng-reflect-view="small"></ish-login-status
    ><ish-mini-basket
      class="d-none d-lg-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-lg-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-lg-none"
      ng-reflect-ng-class="d-lg-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="mobile-logo"
          data-testing-id="header-home-link-mobile"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4"></div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Default Component should render sticky header adequately for tablet devices 1`] = `
<div class="header container tablet" ng-reflect-ng-class="tablet">
  <div class="header-utility">
    <a class="search-toggler active-search"
      ><fa-icon class="header-icon" ng-reflect-icon="fas,search"></fa-icon>
      <div class="search-container header-search-container" ng-reflect-collapsed="false">
        <ish-lazy-search-box data-testing-id="search-box-desktop"></ish-lazy-search-box></div
    ></a>
    <div class="language-switch-container d-none d-md-block"></div>
    <ish-login-status view="small" ng-reflect-view="small"></ish-login-status
    ><ish-mini-basket
      class="d-none d-lg-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-lg-block"
      ng-reflect-view="full"
    ></ish-mini-basket
    ><ish-mini-basket
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-lg-none"
      ng-reflect-ng-class="d-lg-none"
      ng-reflect-view="small"
    ></ish-mini-basket>
  </div>
  <div class="row align-items-center">
    <div class="col-md-3 logo-col">
      <div class="logo-wrapper">
        <a
          class="mobile-logo"
          data-testing-id="header-home-link-mobile"
          rel="home"
          routerlink="/home"
        ></a>
      </div>
    </div>
    <div class="col-md-5 text-right"><ish-lazy-header-quickorder></ish-lazy-header-quickorder></div>
    <div class="col-md-4"></div>
  </div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation></ish-header-navigation
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;
